gomock/call.go

```diff
@@ -435,12 +435,40 @@ func (c *Call) call() []func([]any) []any {
 }
 
 // InOrder declares that the given calls should occur in order.
-func InOrder(calls ...*Call) {
+// Ignores every argument that isn't a *Call or wraps one.
+func InOrder(args ...any) {
+        var calls []*Call
+        for i := 0; i < len(args); i++ {
+                if call := getCall(args[i]); call != nil {
+                        calls = append(calls, call)
+                }
+        }
```

Comment:
we should panic here. Silently passing through this isn't going to help -- clearly indicate the expected parameter type should be either a `*Call` type or a generated mock type which should already embed `*Call`.